feat: Add webhook parser for Sendinblue

ShaneMaglangit opened the pull request:
**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Implemented `getMessageId` and `parseEventBody` to parse Sendinblue webhook.

**Why was this change needed?** (You can also link to an open issue here)

Fixes #1463 

**Other information**:

JSON Response [(Reference)](https://developers.sendinblue.com/docs/how-to-use-webhooks) -- Structure reflects real response (tested by triggering real webhook through the platform)
```json
{
  "event": "delivered",
  "email": "example@example.com",
  "id": 26224,
  "date": "YYYY-MM-DD HH:mm:ss",
  "ts": 1598634509,
  "message-id": "<xxxxxxxxxxxx.xxxxxxxxx@domain.com>",
  "ts_event": 1598034509,
  "subject": "Subject Line",
  "tag": "[\"transactionalTag\"]",
  "sending_ip": "185.41.28.109",
  "ts_epoch": 1598634509223,
  "tags": [
    "myFirstTransactional"
  ]
}
```

Events [(Reference)](https://developers.sendinblue.com/reference/createwebhook)
![image](https://user-images.githubusercontent.com/53674742/195004597-74dbfca2-0832-430c-8e1a-7b26bdd5c1e5.png)

davidsoderberg commented on providers/sendinblue/src/lib/sendinblue.provider.ts:
```diff
@@ -54,6 +56,75 @@ export class SendinblueEmailProvider implements IEmailProvider {
+      return body.map((item) => item['message-id']);
+    }
+
+    return [body['message-id']];
```
this is the same as messageId that is returned from sendMessage?

ShaneMaglangit replied:
Yes, I confirmed it by manually testing the hook. Here is the screenshot from Beeceptor where the response was sent.

![image](https://user-images.githubusercontent.com/53674742/195010585-29a9f83b-494f-4590-8cdc-b5e3238f56f3.png)

davidsoderberg commented on providers/sendinblue/src/lib/sendinblue.provider.ts:
```diff
@@ -54,6 +56,75 @@ export class SendinblueEmailProvider implements IEmailProvider {
+
+  private getStatus(event: string): EmailEventStatusEnum | undefined {
+    switch (event) {
+      case 'sent':
```
Maybe these can be an internal enum inside of this file to make easy to change in the future?

ShaneMaglangit replied:
I believe that is contradictory to what we have [discussed recently](https://discord.com/channels/895029566685462578/937309213036785744/1029245883499429899).

However, I do agree and much prefer creating a new `SendinblueEventStatusEnum`.

davidsoderberg commented on providers/sendinblue/src/lib/sendinblue.provider.ts:
```diff
@@ -54,6 +56,75 @@ export class SendinblueEmailProvider implements IEmailProvider {
+
+    return {
+      status: status,
+      date: body.date,
```
What format do this date have?

ShaneMaglangit replied:
The date was formatted as `YYYY-MM-DD HH:mm:ss`. Is there are standard format that is used for in this? ISO string perhaps?

davidsoderberg commented:
Hi @ShaneMaglangit looks good, would be great if we could test this in the provider spec as well 😄

ShaneMaglangit commented:
@davidsoderberg Hey, I pushed some changes on the PR. Would love to hear your thoughts about it.

I modified the event parsing function based to serve more as a ["dedicated mapping function from provider to novu events"](https://discord.com/channels/895029566685462578/937309213036785744/1029280039864373248), or at least the way that I understood it.

I've also added unit tests for the added functions.

scopsy reviewed (approved):
thank you @ShaneMaglangit!
